roofit/roofitcore/src/TestStatistics/LikelihoodWrapper.cxx

```diff
@@ -49,13 +51,40 @@ namespace TestStatistics {
  */
 LikelihoodWrapper::LikelihoodWrapper(std::shared_ptr<RooAbsL> likelihood,
                                      std::shared_ptr<WrapperCalculationCleanFlags> calculation_is_clean)
-   : likelihood_(std::move(likelihood)), calculation_is_clean_(std::move(calculation_is_clean))
+   : LikelihoodWrapper(likelihood, calculation_is_clean, std::make_shared<std::vector<ROOT::Math::KahanSum<double>>>(),
+                       std::make_shared<std::vector<ROOT::Math::KahanSum<double>>>())
 {
    // Note to future maintainers: take care when storing the minimizer_fcn pointer. The
    // RooAbsMinimizerFcn subclasses may get cloned inside MINUIT, which means the pointer
    // should also somehow be updated in this class.
 }
 
+LikelihoodWrapper::LikelihoodWrapper(std::shared_ptr<RooAbsL> likelihood,
+                                     std::shared_ptr<WrapperCalculationCleanFlags> calculation_is_clean,
+                                     std::shared_ptr<std::vector<ROOT::Math::KahanSum<double>>> offsets,
+                                     std::shared_ptr<std::vector<ROOT::Math::KahanSum<double>>> offsets_save)
+   : likelihood_(std::move(likelihood)),
+     calculation_is_clean_(std::move(calculation_is_clean)),
+     component_offsets_(std::move(offsets)),
+     component_offsets_save_(std::move(offsets_save))
```

Comment:
```suggestion
     component_offsets_(offsets ? std::move(offsets) : std::make_shared<std::vector<ROOT::Math::KahanSum<double>>>()),
     component_offsets_save_(offsets_save ? std::move(offsets_save) : std::make_shared<std::vector<ROOT::Math::KahanSum<double>>>())
```
Note that my other suggestions with avoiding the code duplication would mean something like this. Or, you put the `=make_shared...` in the declaration instead of `=nullptr`.

Reply:
See comment below: no more default parameters, so this won't be necessary. In fact, nullptr shared_ptrs can still be passed explicitly, but the solution is not to handle that with a default value internally, because they always need to be shared.